lazy_tensor_core/lazy_tensor_core/csrc/device.h

```diff
@@ -1,50 +1,47 @@
 #pragma once
 
 #include <iostream>
+#include <memory>
 #include <string>
 
-#include "lazy_tensors/computation_client/util.h"
-#include "torch/csrc/lazy/core/hash.h"
+#include <c10/util/Deprecated.h>
 
 namespace torch_lazy_tensors {
 
-enum class DeviceType { CPU, GPU, TPU };
+// Backend can extend it and define their own supported hardware types.
+struct BackendDeviceType {
```

Comment:
@JackCaoG Here is the new design. This aiming of this new BackendDeviceType is to provide an extensible way for backends to pretty print device type while keeping the many-to-one design philosophy of class Device intact and avoid changing too many code.

Why we need to pretty print Device? It's because we have (and want to offer) APIs that allow users to query the current hardware.